signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/aws/AwsKmsClient.java

```diff
@@ -82,6 +100,74 @@ public byte[] sign(final String kmsKeyId, final byte[] data) {
     return kmsClient.sign(signRequest).signature().asByteArray();
   }
 
+  public <R> MappedResults<R> mapKeyList(
+      final Function<KeyListEntry, R> mapper,
+      final Collection<String> tagKeys,
+      final Collection<String> tagValues) {
+    final Set<R> result = ConcurrentHashMap.newKeySet();
+    final AtomicInteger errorCount = new AtomicInteger(0);
+
+    try {
+      kmsClient
+          .listKeysPaginator()
+          .iterator()
+          .forEachRemaining(
+              listKeysResponse ->
+                  listKeysResponse.keys().parallelStream()
+                      .filter(
+                          keyListEntry ->
+                              keyListPredicate(keyListEntry.keyId(), tagKeys, tagValues))
+                      .filter(this::isEnabledSecp256k1Key)
+                      .forEach(
+                          keyListEntry -> {
+                            try {
+                              final R value = mapper.apply(keyListEntry);
+                              result.add(value);
+                            } catch (final Exception e) {
+                              LOG.warn(
+                                  "Failed to map keyListEntry '{}' to requested object type.",
+                                  keyListEntry.keyId(),
+                                  e);
+                              errorCount.incrementAndGet();
+                            }
+                          }));
+    } catch (Exception e) {
+      LOG.error("Unexpected error during Aws mapKeyList", e);
+      errorCount.incrementAndGet();
+    }
+
+    return MappedResults.newInstance(result, errorCount.intValue());
+  }
+
+  private boolean isEnabledSecp256k1Key(final KeyListEntry keyListEntry) {
+    final KeyMetadata keyMetadata =
+        kmsClient
+            .describeKey(DescribeKeyRequest.builder().keyId(keyListEntry.keyId()).build())
+            .keyMetadata();
+    final boolean isEnabled = keyMetadata.keyState() == KeyState.ENABLED;
+    final boolean isSecp256k1 = keyMetadata.keySpec() == KeySpec.ECC_SECG_P256_K1;
+    return isEnabled && isSecp256k1;
+  }
+
+  private boolean keyListPredicate(
```

Comment:
nit: method name could be a little more descriptive, e.g. `keyMatchesTags`

Reply:
Yeah, the name isn't that great. Renaming it.